tests/test_correlatedNoiseDetection.py

```diff
@@ -0,0 +1,103 @@
+#
+# LSST Data Management System
+#
+# Copyright 2017 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+from __future__ import absolute_import, division, print_function
+
+import unittest
+import numpy as np
+import scipy.ndimage
+
+import lsst.afw.image
+import lsst.meas.algorithms
+import lsst.utils.tests
+
+try:
+    type(display)
+except NameError:
+    display = False
+
+
+class CorrelatedNoiseDetectionTestCase(lsst.utils.tests.TestCase):
+    """Test utilities for dealing with correlated noise when running detection."""
+
+    def testMeasureCorrelatationKernel(self):
```

Comment:
Your test here doesn't use correlated values.  I'm wondering if this is a useful test?  I could imagine getting the correlation values correct for this simple case, but missing a bug that would get the correlation wrong when there actually is correlation.

Reply:
Yup, this was just laziness, because I didn't want to work out how to add correlated stationary noise with NumPy.  If you happen to have a recipe for that on hand, I'd love to have it.

Reply:
I gave this a shot, and unfortunately the naive way to do it with NumPy (constructing a full covariance matrix) is totally impractical - if we want enough pixels in the noise image to be able to estimate the kernel, the covariance matrix is too big to even allocate.  I could imagine working around that with sparse matrices, and I'm sure there's a way to do this with an FFT, but at this point it's not looking like it's worth the effort.

Reply:
Yes, FFT is the way to go.  I could potentially give this a shot today.  It is conceptually simple, but it always takes me a while to get all the factors correct.

Reply:
That'd be great.  I'm not in any hurry to merge this, so please don't feel like you need to get it done today.

Would it make sense to just make that a separate ticket (maybe to add more reusable functionality to `afw::math::Random`), and block the merge of this ticket on that one?

Reply:
Yeah, that sounds like a good idea.

Reply:
https://jira.lsstcorp.org/browse/DM-13044